code/game/dna/genes/goon_powers.dm

```diff
@@ -446,6 +446,10 @@
 		var/failed_leap = 0
 		if (istype(target.loc,/turf/))
 
+			if(istype(target.loc, /turf/space))
+				to_chat(target, "<span class='warning'>You're in space. No.</span>")
```

Comment:
How about something with actual flavor, like "You have nothing to leap off of." with a check for catwalks?